templates/vue-starter-template/app/components/FrontendDetailPage.vue

```diff
@@ -0,0 +1,87 @@
+<script setup lang="ts">
+import { getProductName } from "@shopware/helpers";
+
+const props = defineProps<{
+  navigationId: string;
+}>();
+
+const { search } = useProductSearch();
+const { buildDynamicBreadcrumbs, pushBreadcrumb } = useBreadcrumbs();
+const { apiClient } = useShopwareContext();
+const errorDetails = ref();
+
+const { data, error } = await useAsyncData(
+  `cmsProduct${props.navigationId}`,
+  async () => {
+    const responses = await Promise.allSettled([
+      search(props.navigationId, {
+        withCmsAssociations: true,
+        associations: {
+          seoUrls: {},
+        },
+      }),
+      apiClient.invoke("readBreadcrumb get /breadcrumb/{id}", {
+        pathParams: {
+          id: props.navigationId,
+        },
+      }),
+    ]);
+
+    for (const response of responses) {
+      if (response.status === "rejected") {
+        console.error("[FrontendDetailPage.vue]", response.reason.message);
+        errorDetails.value = response.reason.message;
+      }
+    }
+
+    return {
+      productResponse:
+        responses[0].status === "fulfilled" ? responses[0].value : null,
```

Comment:
why do we need `null`? I'd go with the undefined so
```suggestion
      productResponse: responses[0].value,
```

Reply:
This will lead to a type error. https://stackoverflow.com/questions/63783735/type-error-on-response-of-promise-allsettled <- context